TestModels/Resource/Model/resources.smithy

```diff
@@ -83,3 +97,12 @@ structure SimpleResourcesException {
   message: String,
 }
 
+
+@aws.polymorph#reference(resource: MutableResource)
+structure MutableResourceReference {}
+
+@smithy.api#suppress(["MutableLocalStateTrait"])
```

Comment:
TIL about the @suppress trait, nice!

FYI you can customize the identifier of the validation event (either by changing the name of the validator or passing it explicitly to `danger(...)`. We could be more specific and make it `MutableLocalStateTraitUnsound` or something, but I don't feel strongly.

Reply:
I get it. I don't feel that strongly. So I'm going to leave it.